main.cpp

```diff
@@ -0,0 +1,136 @@
+#include <arpa/inet.h>
+#include <bits/stdc++.h>
+#include <cstdlib>
+#include <cstring>
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+class Config {
+public:
+  Config() = default;
+
+  Config(int argc, char *argv[]) {
+    if (argc > 1)
+      m_HOST = argv[1];
+    if (argc > 2)
+      m_PORT = std::stoi(argv[2]);
+    if (argc > 3)
+      m_numberOfSockets = std::stoi(argv[3]);
+    if (argc > 4)
+      m_numberOfSends = std::stoi(argv[4]);
+    if (argc > 5) {
+      m_messages.clear();
+      for (int i = 5; i < argc; i++) {
+        m_messages.push_back(argv[i]);
+      }
+    }
+  }
+
+  Config &operator=(const Config &other) = default;
+
+  [[nodiscard]] int getPort() const { return m_PORT; }
+
+  void setPort(int port) { m_PORT = port; }
+
+  [[nodiscard]] const std::string &getHost() const { return m_HOST; }
+
+  void setHost(const std::string &host) { m_HOST = host; }
+
+  [[nodiscard]] int getNumberOfSockets() const { return m_numberOfSockets; }
+
+  void setNumberOfSockets(int n) { m_numberOfSockets = n; }
+
+  [[nodiscard]] int getNumberOfSends() const { return m_numberOfSends; }
+
+  void setNumberOfSends(int m) { m_numberOfSends = m; }
+
+  [[nodiscard]] const std::vector<const char *> &getMessages() const {
+    return m_messages;
+  }
+
+  void setMessages(const std::vector<const char *> &messages) {
+    Config::m_messages = messages;
+  }
+
+  [[nodiscard]] __socket_type getSocketType() const { return m_socket_type; }
+
+  ~Config() = default;
+
+private:
+  int m_PORT{8080};
+  std::string m_HOST{"192.168.0.12"};
+  int m_numberOfSockets{3};
+  int m_numberOfSends{4};
+  std::vector<const char *> m_messages{"XXX"};
+  __socket_type m_socket_type{SOCK_DGRAM};
+};
+
+void create_sockets(std::vector<int> &socks, const Config &config) {
+  socks.resize(config.getNumberOfSockets());
+  char buf[INET6_ADDRSTRLEN];
+  auto ipVersion = AF_INET;
+  if (inet_pton(AF_INET6, config.getHost().c_str(), buf)) {
+    ipVersion = AF_INET6;
+  }
+  for (auto &s : socks) {
+    if ((s = socket(ipVersion, config.getSocketType(), 0)) < 0) {
```

Comment:
Don't use "=" operator in conditions. Move up it to make code more readable. It hasn't any program performance effect.